hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2221,11 +2236,17 @@ protected void setDefaults() {
           HoodiePayloadConfig.newBuilder().fromProperties(writeConfig.getProps()).build());
       writeConfig.setDefaultOnCondition(!isMetadataConfigSet,
           HoodieMetadataConfig.newBuilder().withEngineType(engineType).fromProperties(writeConfig.getProps()).build());
-      writeConfig.setDefaultOnCondition(!isLockConfigSet,
-          HoodieLockConfig.newBuilder().fromProperties(writeConfig.getProps()).build());
       writeConfig.setDefaultOnCondition(!isPreCommitValidationConfigSet,
           HoodiePreCommitValidatorConfig.newBuilder().fromProperties(writeConfig.getProps()).build());
       writeConfig.setDefaultValue(TIMELINE_LAYOUT_VERSION_NUM, String.valueOf(TimelineLayoutVersion.CURR_VERSION));
+
+      if (!isLockConfigSet) {
+        HoodieLockConfig.Builder lockConfigBuilder = HoodieLockConfig.newBuilder().fromProperties(writeConfig.getProps());
```

Comment:
HoodieWriteConfig is what we use even during table service schedule and execution. Could we think thru a little on whether we could accidentally disable a configured lock provider in any of those scenarios

Reply:
Clustering and Compaction CLI commands use `UtilHelpers::createHoodieClient()` to create the write client  and the write config constructed there doesn't provide any lock provider. Not sure if these commands will be called with explicit properties option for the lock provider. So, with the default command options, these commands will start to use InProcessLockProvider instead of ZooKeeper based one. Sounds like a problem to me. What do you think? Do people generally provider extra lock provider props when running these commands?

Reply:
Btw, the default write concurrency mode is single writer. Which means, unless these table services are started with  `optimistic_concurrency_control`, these services are not going to get the default lock provider ZooKeeperBasedLockProvider. So, this PR doesn't change the existing flow.

Reply:
>Sounds like a problem to me. What do you think? Do people generally provider extra lock provider props when running these commands?

Well, are we not saying that for metadata table to work, in async deployment modes, people need to turn on a lock provider

Reply:
Right, for the async deployment modes, users must configure the lock provider. And whenever a lock provider is configured, it takes the precedence over any defaults we set.

Reply:
Thinking out aloud. `isLockConfigSet` is only true when the builder method is used. https://github.com/apache/hudi/blob/be47309da1b063283af0365199320b9785046bf5/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java#L2021 I think most of the times, through these tools, the configs would be passed from command line or properties file ..

So those would correctly override what we set in the if block?

Reply:
cc @codope @yihua can one of you take over here?

Reply:
I don't think `isLockConfigSet` is effective. That builder method is not invoked anywhere other than test classes. Should we check for lock provider class name?

Reply:
@codope @vinothchandar addressed your comment and updated the test to cover this case.